easyorder/backend/views.py

```diff
@@ -36,14 +37,55 @@ def dish_list(request):
         return JsonResponse(serializer.errors, status=400)
 
 @csrf_exempt
-def send_notification(request):
-    # TODO: send notification to all users (or some users)
-    pass
+def notification_content_with_timestamp(request, timestamp):
+    '''
+    Get the notification if there is a new one
+    '''
+    try:
+        notif = Notification.objects.all().first()
+    except IndexError:
+        return JsonResponse({'Message':'Server errors'}, status=500)
+
+    if request.method == 'GET':
+        last = datetime.datetime.fromtimestamp(int(timestamp))
+        timezone = notif.modified_at.tzinfo
+        last = last.replace(tzinfo=timezone)
+
+        if last < notif.modified_at:
+            data = NotificationSerializer(notif).data
+            data['notification'] = True
+            data['modified_at'] = format(notif.modified_at, 'U')
+            return JsonResponse(data)
+        else:
+            return JsonResponse({'notification':False})
+
+    return JsonResponse({'Message':'Method Not Allowed'}, status=405)
 
 @csrf_exempt
-def poll_notification(request):
-    # TODO: check whether there is any new notification form the retailer
-    pass
+def notification_content(request):
+    '''
+    Update/Get the notification
+    '''
+    try:
+        notif = Notification.objects.all().first()
+    except IndexError:
+        return JsonResponse({'Message':'Server errors'}, status=500)
+
+    if request.method == 'GET':
+        data = NotificationSerializer(notif).data
+        data['notification'] = True
+        data['modified_at'] = format(notif.modified_at, 'U')
+        return JsonResponse(data)
+    elif request.method == 'PUT':
+        data = JSONParser().parse(request)
+        serializer = NotificationSerializer(notif, data=data)
+
+        if serializer.is_valid():
+            serializer.save()
```

Comment:
understand your idea. just change Notification to derive from SingletonModel in models.py as a constraint